### PR TITLE
A simple test for io arguments

### PR DESCRIPTION
**Pull request by logogin**
More tests to follow.

### CODE REVIEW

File: requirements.dev.txt [hunk 1]
```diff
@@ -4,3 +4,7 @@ sphinx-autodoc-typehints
 recommonmark
 pypandoc
 twine
+
```
Comment (tdomhan) [inline requirements.dev.txt]: extra newline?

File: requirements.dev.txt [hunk 2]
```diff
@@ -4,3 +4,7 @@ sphinx-autodoc-typehints
...
+pytest
```
Comment (tdomhan) [inline requirements.dev.txt]: these test requirements should go to tests_require in setup.py (and are actually already there). The requirements here are additional requirements on top of that.

File: requirements.dev.txt (hunk 1 above)
Comment (logogin) [inline requirements.dev.txt]: should I remove it?

File: requirements.dev.txt (hunk 2 above)
Comment (logogin) [inline requirements.dev.txt]: if I understand correctly `requirements.dev.txt` is for development, so pytest etc. should be declared there. Normally you don't run `setup.py` when import the project into IDE. Also, I am not certain if `setup.py` installs into virtualenvs.

File: requirements.dev.txt (hunk 1 above)
Comment (tdomhan) [inline requirements.dev.txt]: yes :)

File: requirements.dev.txt (hunk 2 above)
Comment (tdomhan) [inline requirements.dev.txt]: it's actually a bit of a misnomer. Currently it's used by the document build server to install dependencies needed for generating the documents. Generally the idea was for it to hold anything that is not a direct project dependency or a test dependency, which currently means doc generation dependencies. Maybe we should rename it to requirements.doc.txt.

Now when running setup.py tests all test dependencies will be installed automatically. This should also include virtualenvs. However, IDEs may or may not parse the setup.py for tests_require.

Anyway, not sure what the best setup is. Maybe a requirements.doc.txt for the documentation build server and a requirements.dev.txt with test dependencies (and maybe twine). Could we leave requirements.dev.txt as is for now and then separately decide whether to split it up into doc/dev?

Comment (logogin) [inline requirements.dev.txt]: ok, I will revert the change to `requirements.dev.txt` for now

**Comment by logogin** [comment]
rebased on top of master

**Pull request merged**